core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -1012,6 +1013,16 @@ private void handleAdditionalProperties(List<String> command) {
                     }
                 }
             }
+
+            private void addExperimentalVMOption(List<String> nativeImageArgs, String option) {
+                if (graalVMVersion.compareTo(GraalVM.Version.VERSION_23_1_0) >= 0) {
+                    nativeImageArgs.add("-H:+UnlockExperimentalVMOptions");
+                }
+                nativeImageArgs.add(option);
+                if (graalVMVersion.compareTo(GraalVM.Version.VERSION_23_1_0) >= 0) {
+                    nativeImageArgs.add("-H:-UnlockExperimentalVMOptions");
+                }
```

Comment:
So you're surrounding the experimental option with enabling/disabling unlock? Is it really how it works? Never seen something similar :).

Reply:
Correct, this is to limit the scope as suggested in https://github.com/oracle/graal/issues/7105#issue-1833493484. Alternatively we could just pass `-H:+UnlockExperimentalVMOptions` as the first option and allow any experimental option to be passed afterwards. It would keep things simple but would also allow people to use experimental options possibly without noticing.